### PR TITLE
test technical documentation

### PR DESCRIPTION
**Pull request by cjdcordeiro**
fixes https://github.com/canonical/rockcraft/issues/54.

In this PR:
 - restructure the docs to make them inherit the code-blocks from tested Spread tasks
 - make Spread look into the docs code to test all documentation tutorials and how-to guides
 - update READMEs
 - simplify some of the python dependencies for the docs
 - add `copybutton` to code-blocks

**NOTE:** for the `docs/how-to/code`, Spread will only run for one system (Jammy).

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Comment by cjdcordeiro** [comment]
(moving back to WIP because I will take the opportunity and also add any tests resulting from the Chisel PRs that are about to be merged)

**Comment by cjdcordeiro** [comment]
this is now ready for review

File: spread.yaml [hunk 1]
```diff
@@ -74,7 +79,7 @@ prepare: |
 restore-each: |
   # Cleanup after each task.
   docker system prune -a -f
-  lxc --project=rockcraft stop --all
+  # lxc --project=rockcraft stop --all
```
Comment (tigarmo) [inline spread.yaml:82]: how come?

File: spread.yaml [hunk 2]
```diff
@@ -8,25 +8,30 @@ environment:
 include:
     - tests/
     - tools/
+    - docs/
+    - requirements-doc.txt
+    - requirements-focal.txt
+    - Makefile
+    - rockcraft/
```
Comment (tigarmo) [inline spread.yaml:15]: why is `rockcraft/` necessary?

File: spread.yaml [hunk 3]
```diff
@@ -8,25 +8,30 @@ environment:
...
     location: snapd-spread/us-east1-b
     halt-timeout: 2h
     systems:
-      - ubuntu-18.04-64:
-          workers: 1
-          storage: 40G
-      - ubuntu-20.04-64:
-          workers: 1
-          storage: 40G
+      # - ubuntu-18.04-64:
```
Comment (tigarmo) [inline spread.yaml:23]: are these temporarily disabled?

File: docs/how-to/code/build-docs/task.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  timeout -s SIGINT 7 \
+  make rundocs
+  # [docs:make-rundocs-end]
+  ret=$?
+  set -e
+
+  if [ $ret -ne 124 ]
+  then
+    exit $ret
+  fi
```
Comment (tigarmo) [inline docs/how-to/code/build-docs/task.yaml:39]: can you please add a newline at the end of these files? Many of them are missing it and github complains

File: docs/how-to/code/create-slice/task.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# as snippets in the docs.
+###########################################
+summary: test the "Create a package slice for Chisel" guide
+
+execute: |
+  # [docs:apt-show-openssl]
+  apt show openssl
+  # [docs:apt-show-openssl-end]
+
+  git clone -b fix-copyright-symlink https://github.com/woky/chisel
```
Comment (tigarmo) [inline docs/how-to/code/create-slice/task.yaml:15]: :/ pity

**Review by tigarmo — CHANGES_REQUESTED** [review]

**Comment by tigarmo** [comment]
Thanks for taking the time to do this, it looks like a lot of very annoying shuffling of code hehe

File: spread.yaml (hunk 3 above)
Comment (cjdcordeiro) [reply spread.yaml:23]: same as above...forgot to undo my local test configurations. fixing it

File: spread.yaml (hunk 1 above)
Comment (cjdcordeiro) [reply spread.yaml:82]: that's a leftover. I do this locally cause for some reason, my spread tests fail to stop LXC instances. fixing it

**Comment by cjdcordeiro** [comment]
> Thanks for taking the time to do this, it looks like a lot of very annoying shuffling of code hehe

:p ty for the review

I've committed the requested changes

File: docs/how-to/code/create-slice/task.yaml (hunk 5 above)
Comment (tigarmo) [reply docs/how-to/code/create-slice/task.yaml:15]: what do you think of pining the Chisel commit to an early (pre-issue) one, like we do for the [rockcraft snap](https://github.com/canonical/rockcraft/blob/main/snap/snapcraft.yaml#L107) itself?

File: docs/how-to/code/build-docs/task.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
```
Comment (tigarmo) [inline docs/how-to/code/build-docs/task.yaml:40]:
```suggestion
```

File: docs/how-to/publish-slice.rst [hunk 7]
```diff
@@ -8,28 +8,29 @@ You have also tested this custom Chisel release, and it works! You believe there
...
-#. Fork this repository https://github.com/canonical/chisel-releases and clone your fork:
+1. Fork this repository https://github.com/canonical/chisel-releases and clone your fork:
 
-    .. code-block:: bash
+.. code-block:: bash
 
-        # Let's assume we are working with Ubuntu 22.04
-        $ git clone -b ubuntu-22.04 https://github.com/<your_github_username>/chisel-releases.git
+    # Let's assume we are working with Ubuntu 22.04
+    git clone -b ubuntu-22.04 https://github.com/<your_github_username>/chisel-releases.git
```
[the remark this answers is not in the document]
Comment (cjdcordeiro) [reply docs/how-to/publish-slice.rst:16]: updated for consistency

**Comment by cjdcordeiro** [comment]
note the new Spread error:
```
+ rockcraft pack
Launching instance...
craft-providers error: Incompatible base detected: Expected OS 'Ubuntu', found None.
```

File: docs/how-to/code/create-slice/task.yaml [hunk 8]
```diff
@@ -12,11 +12,12 @@ execute: |
   apt show openssl
   # [docs:apt-show-openssl-end]
 
-  git clone -b fix-copyright-symlink https://github.com/woky/chisel
+  git clone https://github.com/canonical/chisel
   git clone -b ubuntu-22.04 https://github.com/canonical/chisel-releases
   cp openssl.yaml chisel-releases/slices
```
Comment (tigarmo) [inline docs/how-to/code/create-slice/task.yaml:17]: I guess this `cp` is no longer necessary?

**Review by tigarmo — APPROVED** [review]
looks good, thanks!

File: docs/how-to/code/create-slice/task.yaml (hunk 8 above)
Comment (tigarmo) [reply docs/how-to/code/create-slice/task.yaml:17]: oh of course; I had it in my head that this was part of the copyright symlink fix somehow
thanks

**Review by cmatsuoka — APPROVED** [review]

**Pull request merged**